src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlInternalConnectionTds.cs

```diff
@@ -479,6 +481,11 @@ internal SqlConnectionTimeoutErrorInternal TimeoutErrorInternal
                 _accessTokenInBytes = System.Text.Encoding.Unicode.GetBytes(accessToken);
             }
 
+            if (accessTokenCallback != null)
+            {
+                _accessTokenCallback = accessTokenCallback;
+            }
```

Comment:
The null check doesn't seem necessary here.
```suggestion
            _accessTokenCallback = accessTokenCallback;
```